### PR TITLE
Add Azure Active Directory auth to Back Office

### PR DESCRIPTION
**Pull request by zackads**
## Context

- Requirement for secure access to backoffice tool by Beacons Registry staff requires ability to authenticate with Azure Active Directory.

## Changes in this pull request

Core auth stuff:
- Use official MSFT https://www.npmjs.com/package/@azure/msal-react SDK package to create auth components for application.
- Configure temp Azure AD tenant in my own free tier pending a 'better' tenant (MCA / Made Tech).
- Wrap application in created auth components.

Other:
- Removes Cypress following conversation at dev standup today.
- Adds a pre-push hook to run unit tests before push in order to tighten feedback loop.

## Guidance to review

Merging this PR would mean users will have to authenticate with any Microsoft account in order to see the back office application.  Linda and the team would have to accept this warning:

image

Is this is OK?  I have tested with my MCGA account and it's working.

I can also configure Azure AD to only allow sign-ins from MCGA accounts, or accounts on a pre-determined list, though haven't set this (pending [conversation](https://teams.microsoft.com/l/message/19:e536d0d9c24247b298c9457bc8a02c58@thread.tacv2/1619604181488?tenantId=3fd408b5-82e6-4dc0-a36c-6e2aa815db3e&groupId=f4fd0372-f10b-410f-9491-eac811ac53dd&parentMessageId=1619604181488&teamName=The%20Beacon%20Registry%20Project&channelName=Development%20Channel&createdTime=1619604181488) with @paulgutman)

Relevant docs:
- https://www.npmjs.com/package/@azure/msal-react 

## Link to Trello card

- N/A, cross-functional

## Things to check

- [X ] Infrastructure - _N/A_
- [X] Environment variables - _N/A_

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,18 +10,17 @@
...
-    "cypress:watch": "cypress open --config watchForFileChanges=true",
-    "cypress:run": "cypress run",
-    "cypress:run:firefox": "cypress run --browser firefox"
+    "format:check": "prettier --check '**/*.{js,jsx,json,ts,tsx,yml,md,html,scss,css}'"
   },
   "husky": {
     "hooks": {
-      "pre-commit": "lint-staged"
+      "pre-commit": "lint-staged",
+      "pre-push": "git diff HEAD --quiet && npm run test -- --watchAll=false"
```
Comment (zackads) [inline package.json:18]: `git diff HEAD --quiet` ensures that local un-committed files do not result in unit tests passing, which then fail in CI with only the committed files.  Trying to push with locally un-committed files will error.  (See https://faun.pub/why-using-pre-push-git-hooks-with-husky-is-not-always-a-good-idea-6233b8afcf83)

File: src/components/auth/AuthWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React, { FunctionComponent } from "react";
+
+import { MsalProvider } from "@azure/msal-react";
+import { Configuration, PublicClientApplication } from "@azure/msal-browser";
+
+const configuration: Configuration = {
+  auth: {
+    clientId: "276fd164-c30c-485f-bd76-a8f3b5bcac74",
```
Comment (zackads) [inline src/components/auth/AuthWrapper.tsx:8]: `clientId` is not a secret and so is OK to keep in VCS (I believe).  See [OAuth specification](https://tools.ietf.org/html/rfc6749#section-2.2) and [StackOverflow](https://stackoverflow.com/questions/61613985/are-azure-application-client-ids-safe-to-store-in-the-frontend).

Deliberately chose to keep the auth configuration with the `AuthWrapper` component (i.e. not separate into `authConfig.json` or similar) so there is one place for all things 'auth'.

**Review by NyWeb — APPROVED** [review]
This all looks good to me. Personally, I would rip out Cypress with a different PR (or no PR) to not muddy the waters and make reviews easier, but I get your point of merging them too.

File: package.json (hunk 1 above)
Comment (matthew-a-carr) [reply package.json:18]: @zackads @Ninamma 

Sorry to come late to the party on this. I'd find this personally quite annoying and would prefer to have a separate terminal open with `npm run test:watch` running. 

Running the tests on every push could also become frustrating when the test suit grows and takes > 5 seconds to run

**Pull request merged**